elyra/pipeline/airflow/processor_airflow.py

```diff
@@ -636,18 +639,59 @@ def render_executor_config_for_custom_op(op: Dict) -> Dict[str, Dict[str, List]]
 
         :returns: a dict defining the volumes and mounts to be rendered in the DAG
         """
-        executor_config = {"KubernetesExecutor": {"volumes": [], "volume_mounts": []}}
-        for volume in op.get("volumes", []):
-            # Define volumes and volume mounts
-            executor_config["KubernetesExecutor"]["volumes"].append(
-                {
-                    "name": volume.pvc_name,
-                    "persistentVolumeClaim": {"claimName": volume.pvc_name},
-                }
-            )
-            executor_config["KubernetesExecutor"]["volume_mounts"].append(
-                {"mountPath": volume.path, "name": volume.pvc_name, "read_only": False}
-            )
+
+        executor_config = {"KubernetesExecutor": {}}
+
+        # Handle volume mounts
+        if op.get("volumes"):
+            executor_config["KubernetesExecutor"]["volumes"] = []
+            executor_config["KubernetesExecutor"]["volume_mounts"] = []
+            for volume in op.get("volumes", []):
+                # Add volume and volume mount entry
+                executor_config["KubernetesExecutor"]["volumes"].append(
+                    {
+                        "name": volume.pvc_name,
+                        "persistentVolumeClaim": {"claimName": volume.pvc_name},
+                    }
+                )
+                executor_config["KubernetesExecutor"]["volume_mounts"].append(
+                    {"mountPath": volume.path, "name": volume.pvc_name, "read_only": False}
+                )
+
+        # Handle annotations
+        if op.get("kubernetes_pod_annotations"):
+            executor_config["KubernetesExecutor"]["pod_override"] = []
+            for annotation in op.get("kubernetes_pod_annotations", []):
+                # Add Kubernetes annotation entry
+                executor_config["KubernetesExecutor"]["pod_override"].append(
+                    {
+                        "key": annotation.key,
+                        "value": annotation.value,
+                    }
+                )
+
+        return executor_config
+
+    @staticmethod
+    def render_executor_config_for_generic_op(op: Dict) -> Dict[str, Dict[str, List]]:
```

Comment:
Should this also take an `executor_config: Dict` parameter and then we could just call it in `render_executor_config_for_custom_op()`?  What is unique about a generic operation here?

However, I think what would be ideal is to extend the `@dataclass` objects for these "add-ons" to be richer (and siblings of the same base class) where we could also hang all of the validation logic.   There are multiple patterns forming here that I think could be eliminated (or condensed) - validation, "rendering", etc.

If items are specific to a platform, the processor_xxx for platform xxx could use subclasses specific to xxx and apply those methods there - so that platform specifics don't need to be included in the "base" classes.

Reply:
> However, I think what would be ideal is to extend the @dataclass objects for these "add-ons" to be richer (and siblings of the same base class) where we could also hang all of the validation logic. There are multiple patterns forming here that I think could be eliminated (or condensed) - validation, "rendering", etc.

You've read my mind - I've got such changes in the works 🙂

Reply:
> Should this also take an executor_config: Dict parameter and then we could just call it in render_executor_config_for_custom_op()? What is unique about a generic operation here?

The `KubernetesPodOperator` supports certain parameters natively that other operators don't. Therefore the `executor_config` isn't necessarily the same for generic and custom components.

Reply:
I understand and agree that the overall methods should remain separate.  What I'm getting at is that L661-L666 == L679-L684 and, at that granularity, we shouldn't duplicate things - which led to the comment about converting the `@dataclass` instances to actual class instances.

Reply:
I guess a general overhaul of the Airflow code generation in an upcoming release might be the best path forward as your feedback also applies to https://github.com/elyra-ai/elyra/pull/2848. As is, this PR and 2848 will result in a ton of merge conflicts that still need to be sorted out and we have only a few days left before the planned release date. 

Also it appears that the current code generation for volumes unnecessarily differentiates between generic and custom ops. Both can utilize the `executor_config` to define mounts/volumes, resulting in the removal of a non-trivial amount of code.

wdyt?

Reply:
Sounds good.  We can circle back on this when time permits.